### PR TITLE
Add missing positions on trees and add a position checker

### LINKED ISSUE #316: Missing positions on VCs
A lot of VCs are currently missing positions, so I suggest adding a `--debug=position` flag that will enable a checking phase after every `ExtractionPipeline` that will walk the trees while keeping track of the last know position, and report missing positions on every (non-synthetic) tree.

What do you think, @samarion?

### PR DESCRIPTION
**Pull request by romac**
Fix #316

### CODE REVIEW

File: core/src/main/scala/stainless/ast/SymbolOps.scala [hunk 1]
```diff
@@ -260,23 +262,27 @@ trait SymbolOps extends inox.ast.SymbolOps { self: TypeOps =>
...
     * where the condition can be added to the expression in a position
-    * that implies further positions.
+    * that implies further assertions.
+    * 
+    * The last argument `pos` is used to give a proper position to the
+    * synthetic boolean literal `true` that is used as a base case.
+    * Without it, we would lose position information in the resulting tree.
     */
-  def withShared(path: Path, es: Seq[Expr], recons: (Expr, Seq[Expr]) => Expr): Expr = {
+  def withShared(path: Path, es: Seq[Expr], recons: (Expr, Seq[Expr]) => Expr, pos: Position): Expr = {
```
Comment (romac) [inline core/src/main/scala/stainless/ast/SymbolOps.scala]: This is ugly and a bit of a hack, suggestions welcome! @samarion

**Comment by jad-hamza** [comment]
This one can be closed if https://github.com/epfl-lara/stainless/pull/308 gets merged

**Comment by samarion** [comment]
See my comment at https://github.com/epfl-lara/stainless/pull/342#issuecomment-423808222, I think the same approach can be used for the position checker.

**Comment by romac** [comment]
Superseded by #342.

**Pull request closed without merging**